EIPS/eip-5023.md

```diff
@@ -0,0 +1,143 @@
+
+
+
+## Backwards Compatibility
+
+TBD
+
+## Reference Implementation
+
+Following reference implementation demonstrates a general use case of one of our pilots. In this case a shareable non-transferable token represents a contribution done to a community that the contract owner has decided to merit with a token. Contract owner can mint a merit token and give it to a person. This token can be further shared by the receiver to other parties for example to share the received merit to others that have participated or influenced his contribution.
+
+```
+// SPDX-License-Identifier: CC0
+pragma solidity ^0.8.0;
+
+import "./IERCsntNFT.sol";
+import "@openzeppelin/contracts/token/ERC721/IERC721.sol";
+import "@openzeppelin/contracts/token/ERC721/IERC721Receiver.sol";
+import "@openzeppelin/contracts/utils/Address.sol";
+import "@openzeppelin/contracts/utils/Context.sol";
+import "@openzeppelin/contracts/utils/Strings.sol";
+import "@openzeppelin/contracts/utils/introspection/ERC165.sol";
+import "@openzeppelin/contracts/token/ERC721/extensions/IERC721Metadata.sol";
+import "@openzeppelin/contracts/token/ERC721/extensions/ERC721URIStorage.sol";
+import "@openzeppelin/contracts/access/Ownable.sol";
+
+contract ShareableERC721 is ERC721URIStorage, Ownable, IERCsntNFT {
+
+  string baseURI;
+    
+  constructor(string memory _name, string memory _symbol) ERC721(_name, _symbol) {}
+
+  function mint(
+        address account,
+        uint256 tokenId
+    ) external onlyOwner {
+        _mint(account, tokenId);
+  }
+
+  function setTokenURI(
+        uint256 tokenId, 
+        string memory tokenURI
+    ) external {
+        _setTokenURI(tokenId, tokenURI);
+  }
+
+  function setBaseURI(string memory baseURI_) external {
+        baseURI = baseURI_;
+  }
+    
+  function _baseURI() internal view override returns (string memory) {
+        return baseURI;
+  }
+
+  function share(address to, uint256 tokenIdToBeShared, uint256 newTokenId) external virtual {
+      require(to != address(0), "ERC721: mint to the zero address");
+      require(_exists(tokenIdToBeShared), "ShareableERC721: token to be shared must exist");
+      require(!_exists(newTokenId), "token with given id already exists");
+      
+      require(msg.sender == ownerOf(tokenIdToBeShared), "Method caller must be the owner of token");
+
+      string memory _tokenURI = tokenURI(tokenIdToBeShared);
+      _mint(to, newTokenId);
+      _setTokenURI(newTokenId, _tokenURI);
+
+      emit Share(msg.sender, to, newTokenId, tokenIdToBeShared);
+  }
+
+  function transferFrom(
+        address from,
+        address to,
+        uint256 tokenId
+    ) public virtual override {
+        revert('In this reference implementation tokens are not transferrable');
+    }
+
+    function safeTransferFrom(
+        address from,
+        address to,
+        uint256 tokenId
+    ) public virtual override {
+        revert('In this reference implementation tokens are not transferrable');
+    }
+}
```

Comment:
Are you missing `supportsInterface`?

Reply:
Use of supportsInterface is implied. I can add it to the reference implementation if you find it useful.

Reply:
Could add a `/* EIP155 */` comment in the `contract ... is` line?